sdk/iOS/test/MSClientTests.m

```diff
@@ -145,27 +101,58 @@ -(void) testInitWithApplicationURLAllowsNilURL
     XCTAssertNil(client.applicationKey, @"client.applicationKey should be nil.");
 }
 
--(void) testInitWithApplicationURLAndApplicationKey
+
+#pragma mark * Login Method Tests
+
+-(void) testLoginHostSuccess
 {
-    NSURL *appURL = [NSURL URLWithString:@"http://someURL.com"];
+    MSClient *client = [[MSClient alloc] initWithApplicationURL:[NSURL URLWithString:@"http://someURL.com"]];
+    XCTAssertEqualObjects(client.loginHost.absoluteString, @"https://someURL.com");
     
-    MSClient *client =
-    [[MSClient alloc] initWithApplicationURL:appURL
-                    applicationKey:@"here is some key"];
+    client.loginHost = [NSURL URLWithString:@"https://anotherURL.com"];
+    XCTAssertEqualObjects(client.loginHost.absoluteString, @"https://anotherURL.com");
+}
+
+-(void) testLoginHostEncodingSuccess
+{
+    MSClient *client = [MSClient clientWithApplicationURLString:@"http://yeah! .com"];
+    XCTAssertEqualObjects(client.loginHost.absoluteString, @"https://yeah!%20.com");
     
-    XCTAssertNotNil(client, @"client should not be nil.");
+    client.loginHost = [NSURL URLWithString:@"https://no!%20.com"];
+    XCTAssertEqualObjects(client.loginHost.absoluteString, @"https://no!%20.com");
+}
+
+-(void) testLoginHostWithPaths
+{
+    MSClient *client = [[MSClient alloc] initWithApplicationURL:[NSURL URLWithString:@"http://someURL.com/mypath/junk"]];
+    XCTAssertEqualObjects(client.loginHost.absoluteString, @"https://someURL.com");
+    
+    @try {
+        client.loginHost = [NSURL URLWithString:@"https://anotherURL.com/mypath/junk"];
+        XCTAssert(@"Error should have been thrown");
+    }
+    @catch (NSException *exception) {
+        XCTAssertEqualObjects(exception.name, @"InvalidLoginHost");
+    }
+    
+    XCTAssertEqualObjects(client.loginHost.absoluteString, @"https://someURL.com");
+}
+
+-(void) testLoginPrefixSuccess
+{
+    MSClient *client = [[MSClient alloc] initWithApplicationURL:[NSURL URLWithString:@"http://someURL.com/mypath/junk"]];
+    XCTAssertEqualObjects(client.loginPrefix, @".auth/login");
     
-    XCTAssertNotNil(client.applicationURL, @"client.applicationURL should not be nil.");
-    NSString *urlString = [client.applicationURL absoluteString];
-    XCTAssertTrue([urlString isEqualToString:@"http://someURL.com"],
-                 @"The client should be using the url it was created with.");
+    client.loginPrefix = @"login";
+    XCTAssertEqualObjects(client.loginPrefix, @"login");
+
+    client.loginPrefix = @".auth/login";
+    XCTAssertEqualObjects(client.loginPrefix, @".auth/login");
     
-    XCTAssertNotNil(client.applicationKey, @"client.applicationKey should not be nil.");
-    XCTAssertTrue([client.applicationKey isEqualToString:@"here is some key"],
-                 @"The client should be using the url it was created with.");
+    client.loginPrefix = @".auth/ space/!@#";
+    XCTAssertEqualObjects(client.loginPrefix, @".auth/%20space/!@%23");
 }
```

Comment:
updated structure slightly so loginURL can be tested this way and see how they combine in the test since Login and Controller aren't unit tested atm.